src/diem/testing/miniwallet/app/offchain_api_v2.py

```diff
@@ -0,0 +1,200 @@
+# Copyright (c) The Diem Core Contributors
+# SPDX-License-Identifier: Apache-2.0
+
+from dataclasses import asdict
+from typing import Dict
+from .store import NotFoundError
+from .models import Account, Subaddress, PaymentCommand, Transaction
+from .app import App
+from .... import jsonrpc, offchain, utils
+from ....offchain import Status, AbortCode, CommandResponseObject, CommandRequestObject
+
+
+class OffChainAPIv2:
+    def __init__(self, app: App) -> None:
+        self.app = app
+        self.cache: Dict[str, CommandResponseObject] = {}
+
+    def process(self, request_id: str, sender_address: str, request_bytes: bytes) -> CommandResponseObject:
+        try:
+            if not request_id:
+                raise offchain.protocol_error(
+                    offchain.ErrorCode.missing_http_header, "missing %s" % offchain.X_REQUEST_ID
+                )
+            if not offchain.UUID_REGEX.match(request_id):
+                raise offchain.protocol_error(
+                    offchain.ErrorCode.invalid_http_header, "invalid %s" % offchain.X_REQUEST_ID
```

Comment:
could you use format strings here?

Reply:
I'm moving away from it because it is more consistent with logging message to use placeholder and format.